salt/payload.py

```diff
@@ -164,7 +163,9 @@ def ext_type_encoder(obj):
             return tuple(obj)
         elif isinstance(obj, CaseInsensitiveDict):
             return dict(obj)
-        elif isinstance(obj, collections.abc.MutableMapping):
+        elif isinstance(obj, salt.loader.context.NamedLoaderContext):
```

Comment:
I'd like us to handle NamedLoaderContext explicitly and continue to handle MutableMapping.

```
elif isinstance(obj, NamedLoaderContext):
    return ext_type_encoder(obj)
elif isinstance(obj, collectons.abc.MutableMapping):
    return dict(obi)

Reply:
Yes, I agree.

Reply:
@dwoz No problem with adding MutableMapping in, but 
```
return ext_type_encoder(obj)
```
is confusing me since that is how the ext_type_encoder is being called now, and need to detect NamedLoaderContext is essentially None and cannot do 'dict(None)'

Can you explain further what you mean by that line

Reply:
`return ext_type_encoder(obj)` will return `None` if `obj is None`

Reply:
If returning `None` causes this patch not to fix the issue then we need to figure out why this change from a dictionary to None and apply the fix further up the stack someplace.

Reply:
I wonder how it ever returned a dictionary, esp. since this is reported since 3003+.  From debug statements
```
121 2023-04-12 09:37:44,362 [salt.payload     :171 ][WARNING ][91707] DGM payload ext_type_encoder, obj instance of collections.abc.MutableMapping, obj '<salt.loader.context.NamedLoaderContext object at 0x7fe128b01310>'
122 2023-04-12 09:37:44,362 [salt.loader.context:70  ][WARNING ][91707] DGM cls NamedLoaderContext value, name '__grains__', loader 'None', default 'None'
```
The default being applied is None and the loader value from loader_context.loader() is None too. 
So should the default for a NamedLoaderContext when it is created really be an empty dict ?

Reply:
Please review changes made after our conversation